widgetssdk/src/main/java/com/glia/widgets/core/callvisualizer/domain/GliaOnCallVisualizerEndUseCase.kt

```diff
@@ -0,0 +1,69 @@
+package com.glia.widgets.core.callvisualizer.domain
+
+import com.glia.androidsdk.Engagement
+import com.glia.androidsdk.omnibrowse.OmnibrowseEngagement
+import com.glia.widgets.callvisualizer.CallVisualizerRepository
+import com.glia.widgets.core.notification.domain.RemoveCallNotificationUseCase
+import com.glia.widgets.core.notification.domain.RemoveScreenSharingNotificationUseCase
+import com.glia.widgets.core.operator.GliaOperatorMediaRepository
+import com.glia.widgets.core.survey.GliaSurveyRepository
+import com.glia.widgets.core.visitor.GliaVisitorMediaRepository
+
+class GliaOnCallVisualizerEndUseCase(
+    private val repository: CallVisualizerRepository,
+    private val operatorMediaRepository: GliaOperatorMediaRepository,
+    private val callVisualizerUseCase: GliaOnCallVisualizerUseCase,
+    private val removeCallNotificationUseCase: RemoveCallNotificationUseCase,
+    private val removeScreenSharingNotificationUseCase: RemoveScreenSharingNotificationUseCase,
+    private val surveyRepository: GliaSurveyRepository,
+    private val gliaVisitorMediaRepository: GliaVisitorMediaRepository
+) : GliaOnCallVisualizerUseCase.Listener {
+
+    interface Listener {
+        fun engagementEnded()
+    }
+
+    inner class EndCallVisualizerRunnable(private val engagement: Engagement) : Runnable {
+        override fun run() {
+            surveyRepository.onEngagementEnded(engagement)
+            if (listener != null) {
+                listener!!.engagementEnded()
+            }
```

Comment:
```suggestion
            listener?.engagementEnded()
```

Reply:
Oh, yes. It's an auto-generated code. I should have changed it. Thank you.

Reply:
Fixed dce16ed0a888aaf626697a7a745ac3a6d4fc19f2